### PR TITLE
Fix stale inline diagnostics (#65571)

### PR DESCRIPTION
**Pull request by akhera99**
Cherry pick of #65571 
Fixes: #65503

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs [hunk 1]
```diff
@@ -194,13 +168,17 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
...
 
+                // This is what places the diagnostic UI at the end of the line of code or at the end of the TextView window.
                 Canvas.SetLeft(visualElement,
-                    tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfCode ? lineView.Right :
-                    tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfEditor ? TextView.ViewportRight - visualElement.DesiredSize.Width :
-                    throw ExceptionUtilities.UnexpectedValue(tag.Location));
+                    tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfCode
+                        ? Math.Max(lineView.Right, lineView.LineTransform.Right)
+                        : tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfEditor ? TextView.ViewportRight - visualElement.DesiredSize.Width
+                        : throw ExceptionUtilities.UnexpectedValue(tag.Location));
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:176]:
```suggestion
                        : tag.Location == InlineDiagnosticsLocations.PlacedAtEndOfEditor
                            ? TextView.ViewportRight - visualElement.DesiredSize.Width
                            : throw ExceptionUtilities.UnexpectedValue(tag.Location));
```

**Comment by CyrusNajmabadi** [comment]
What's different between this and hte last fix?

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi @akhera99 I believe this is just the cherry-pick back to the 17.4 branch?

**Comment by akhera99** [comment]
Closing since we're not taking it in 17.4.

**Pull request closed without merging**